src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,130 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size = 0;
+    private int capacity = DEFAULT_CAPACITY;
+    private Object[] array;
+
+    public ArrayList() {
+        array = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growIfArrayFull();
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        growIfArrayFull();
+        size++;
+        checkBounds(index);
 
+        for (int i = array.length - 1; i > index; i--) {
+            array[i] = array[i - 1];
+        }
+
+        array[index] = value;
```

Comment:
You should rethink of the proper order for these operations.
First, validate the input - what if index bounds are invalid? You're left with altered size. Likewise, is there need to grow the array in such scenario?

FYI: the operation in loop can be replaced with `System.arraycopy` method.